index.js

```diff
@@ -6,6 +6,60 @@ const KEYCODE = {
 }
 
 const state = new Map()
+// when container or children get focus
+const onFocusin = e => {
+  const {currentTarget: rover} = e
+  if (state.get('last_rover') == rover) return
+  if (state.has(rover)) {
+    activate(rover, state.get(rover).active)
+    state.set('last_rover', rover)
+  }
+}
+const onKeydown = e => {
+  const {currentTarget: rover} = e
+  switch (e.keyCode) {
+    case KEYCODE[isRtl ? 'LEFT' : 'RIGHT']:
+    case KEYCODE.DOWN:
+      e.preventDefault()
+      focusNextItem(rover)
+      break
+    case KEYCODE[isRtl ? 'RIGHT' : 'LEFT']:
+    case KEYCODE.UP:
+      e.preventDefault()
+      focusPreviousItem(rover)
+      break
+  }
+}
+const mo = new MutationObserver((mutationList, observer) =>{
+  const stateElementsSet = new Set();
+  state.forEach((v,k) =>{
+    stateElementsSet.add(k.classList[0])
+  } )
+  mutationList
+    .filter(x => x.removedNodes.length > 0)
+    .forEach(mutation => {
+      mutation.removedNodes.forEach(removedEl => {
+      if (removedEl.nodeType !== 1) return //only elements
+        state.forEach((val,key) => {
+          if (key ==='last_rover') return;
+          if (removedEl.contains(key)) {
+            const currentEl = val;
+            key.removeEventListener('focusin', onFocusin)
+            key.removeEventListener('keydown', onKeydown)
+            state.delete(key)
+            currentEl.targets.forEach(a => a.tabIndex = '') 
+            const keys = [...state.keys()]?.filter(x => x!=='last_rover')           
```

Comment:
this is checking to see if there's anything in the state object other than a last_rover entry, and if so you can safely assume there's no rovers on the page? could just check `state.size === 1` right, since the only entry left will be the last_rover? could even check if `state.has('lazt_rover') && state.size === 1`.

Reply:
Yeah I'd have to use the 2nd check because I think the  last_rover is only added if the onFocusIn is called. So the map size could get down to 1 without that key being added if the rovers were created but never used. One of my earlier efforts would have thrown an error but didn't since I didn't call that method in my testing.

Reply:
Thinking it through a little more I don't think the 2nd test would always work either. because state.has('last_rover') could be false. That's why I was using the filter. if it's there ignore it otherwise the state is just rovers.